src/test/java/org/folio/service/mapping/MappingServiceUnitTest.java

```diff
@@ -91,5 +96,45 @@ void shouldMapInstance_to_marcRecord() {
     String expectedMarcRecord = readFileContentFromResources("mapping/expected_marc_record.mrc");
     Assert.assertEquals(expectedMarcRecord, actualMarcRecord);
   }
+
+  @Test
+  void shouldMapInstanceHoldingsAndItem_to_marcRecord_whenMappingProfileTransformationsAreNotEmpty() {
+    // given
+    JsonObject instance = new JsonObject(readFileContentFromResources("mapping/given_inventory_instance.json"));
+    List<JsonObject> instances = Collections.singletonList(instance);
+    MappingProfile mappingProfile = new MappingProfile();
+    mappingProfile.setTransformations(createHoldingsAndItemSimpleFieldTransformations());
+    Mockito.when(referenceDataProvider.get(jobExecutionId, params)).thenReturn(referenceData);
+    // when
+    List<String> actualMarcRecords = mappingService.map(instances, mappingProfile, jobExecutionId, params);
+    // then
+    Assert.assertEquals(1, actualMarcRecords.size());
+    String actualMarcRecord = actualMarcRecords.get(0);
+    String expectedMarcRecord = readFileContentFromResources("mapping/expected_marc_record_with_holdings_and_items.mrc");
+    Assert.assertEquals(expectedMarcRecord, actualMarcRecord);
+  }
+
+  private List<Transformations> createHoldingsAndItemSimpleFieldTransformations() {
+    List<Transformations> transformations = new ArrayList<>();
+    transformations.add(createTransformations("callNumber", "$.holdings[*].callNumber", "900  $a", HOLDINGS));
```

Comment:
Can you please add atleast one scenario of different indicators instead of just blanks, so that everything is tested

Reply:
Agree, will add one more test case with indicators